bioio_base/io.py

```diff
@@ -62,3 +64,38 @@ def pathlike_to_fs(
     # any open file buffers _after_ any API call. API calls must themselves call
     # fs.open and complete their function during the context of the opened buffer.
     return fs, path
+
+
+def search_for_node(
+    parent: ElementTree.Element, tag: str, attributes: typing.Optional[dict] = None
+) -> typing.Optional[ElementTree.Element]:
```

Comment:
I think this utility function needs unit tests.

It might run faster (and also may resolve @kmitcham's question) by using the builtin [findall](https://docs.python.org/3/library/xml.etree.elementtree.html#example). Something like this (untested):

```python
attribute_queries = [ f'[@{name}="{value}"]' for name, value in attributes]
return parent.findall(f".//{tag}{''.join(attribute_queries)})
```